speedy-recovery-client/src/components/secured/home/HomePageForParent.js

```diff
@@ -0,0 +1,51 @@
+import React, { Component } from "react";
+import { Link } from "react-router-dom";
+import { Loader } from "semantic-ui-react";
+
+class HomePageForParent extends Component {
+  render() {
+    if (
+      this.props.user &&
+      this.props.events &&
+      this.getNextEvent(this.props.events).start &&
+      this.getNextEvent(this.props.events).patient
+    ) {
+      return (
+        <div>
+          <p>Hello {this.props.user.name}</p>
+          {new Date(this.getNextEvent(this.props.events).start).toLocaleString(
+            "en-uk"
+          ) !== "Invalid Date" ? (
+            <p>
+              Here is the time for your child{" "}
+              {this.getNextEvent(this.props.events).patient}'s next appointment:{" "}
+              <Link to={"/secured/calendar"}>
+                {this.formatDate(this.getNextEvent(this.props.events).start)}
+              </Link>
+            </p>
+          ) : (
+            <p>You dont't have any appointment</p>
+          )}
+        </div>
+      );
+    } else {
+      return <Loader content="Loading" inline="centered" active size="large" />;
+    }
+  }
+
+  getNextEvent(events) {
```

Comment:
Please extract getNextEvent and formatDate into a service to avoid duplicate code in the 3 views.